### DISCUSSION TITLE
Is there a way to implement Result Cluster using PnP results webpart?

### DISCUSSION THREAD

**Discussion by mw-new-test** [microsoft-search/pnp-modern-search#3212] [Ideas]
In Office/Bing/SharePoint tenant level search, Microsoft has implemented something known as a **Result Cluster** which shows results from a Graph connector within a Cluster. It is placed on the **All** vertical between/in middle of a set of other M365 results.
Below the cluster section, there is a link to the actual vertical of that particular connector.

For us in PnP modern search and as per the ranking model in Graph API, most results shown in 1st to 3rd pages are from external connectors. We would like to restrict the number of results from Graph connectors to a cluster and then allow users to navigate to the connector specific vertical.

Is there a way to implement such feature in PnP Modern Search - results webpart?

**Comment by wobba** [comment]
Technically it should be doable as you would query different sources and put them together in the UX. But this would have to be custom developed. We would have to support like explained here: https://learn.microsoft.com/en-us/graph/search-concept-multiplerequest
